src/main/java/io/cdap/plugin/snowflake/source/batch/SnowflakeSourceAccessor.java

```diff
@@ -77,7 +78,12 @@ public SnowflakeSourceAccessor(SnowflakeBatchSourceConfig config, String escapeC
    */
   public List<String> prepareStageSplits() {
     LOG.info("Loading data into stage: '{}'", STAGE_PATH);
-    String copy = String.format(COMAND_COPY_INTO, QueryUtil.removeSemicolon(config.getImportQuery()));
+    String importQuery = config.getImportQuery();
+    if (Strings.isNullOrEmpty(importQuery)) {
+      String tableName = config.getTableName();
+      importQuery = String.format("SELECT * FROM %s", tableName);
```

Comment:
ideally we should only import columns present in the schema, why `SELECT *`?

Reply:
Refactored the code. b7ca0955ea02ad4dfdef6116bf9b1a65ae5463c5 PTAL.